### ISSUE TITLE
darwinkit: 0.5.0-preview release tasks

### ISSUE THREAD

**Issue by progrium** [progrium/macdriver#177]
Now that the `darwinkit` branch has been merged into `main`, this issue is to track progress towards cutting a `0.5.0-preview` release. This pre-release will bring attention to the major breaking changes of the upcoming `0.5.0` release, solicit more feedback while things are in flux, and announce the intention to rename to DarwinKit with `0.5.0`.

So what are the *benefits* of all this change? Here are some features of this new version:

- Significantly greater generated class bindings coverage (currently 18x more, from ~80 to over 1500)
- Generated enums and constants, currently at around 7k of them
- Simpler, more powerful generation tooling with no dependency on macschema
- Direct support of native Go builtin types like strings, bools, []byte, maps, etc in bindings
- More 1-to-1 mapping to Objective-C symbol names while also improving Go idiomaticness 
- Support for block arguments as Go functions with properly typed arguments (callbacks!)
- Pre-made delegate implementations you can simply set Go functions on
- Documentation for all symbols including a link to official Apple docs on that symbol
- Setting up to support multiple recent macOS versions
- Setting up to support more platforms like iOS, watchOS, maybe visionOS
- No more compile warnings
- More examples
- Closes a lot of open issues :)

Check out all the [new and ported examples](https://github.com/progrium/macdriver/tree/darwinkit/macos/_examples) and let me know they work on your platform. Below is what I'm thinking needs to be done before the preview release is cut. Please let me know if anything else should be added.

## TODO
- [x] write docs on generate toolchain
- [x] finish manual docs on `objc`, `dispatch`, and `kernel` 
- [x] make issues for filling out placeholder structs
- [x] update examples that need updated `objc` API
- [x] make issue for ideal custom overrides
- [x] make issue for repo/module rename plan
- [x] make issue for string/NSString problem
- [x] make issue for form,layout example exception
- [x] finish decoupling more of remaining stubbed frameworks (at least vision)
- [x] https://github.com/progrium/macdriver/issues/181
- [x] https://github.com/progrium/macdriver/issues/186
